### PR TITLE
fundrawtransaction: Keep change-output keys by default, make it optional

### LINKED ISSUE #9362: fundrawtransaction doesn't keep change address pool key
If one calls `fundrawtransaction`, the change address will be retrieved from the keypool, though, this key/address will always be returned to the pool, leading to the problem that the next `getnewaddress()` call will restult in the same key/address.

Either we remove the auto-change-address (users would need to call `getrawchangeaddress` first [current best practice anyway]) or we add a call to keep/mark a specific key/address as change (after one have sent the funded raw transaction).

Reported by @NicolasDorier on IRC.

### PR DESCRIPTION
**Pull request by jonasschnelli**
Alternative solution for #9370, fixes #9362.

This adds `reserveChangeKey`, a new boolean option to `fundrawtransaction`.
The default is `true` resulting in always removing the change output address key from the keypool.

Includes test and documentation in the release-notes.

### CODE REVIEW

**Comment by laanwj** [comment]
Thanks! Concept ACK.
We could still use #9370 in some form after this, but this is a more effective solution to the issue.

**Comment by gmaxwell** [comment]
Concept ACK. Thanks!

File: qa/rpc-tests/fundrawtransaction.py [hunk 1]
```diff
@@ -660,5 +660,32 @@ def run_test(self):
...
+        nextaddr = self.nodes[3].getnewaddress()
+        # frt should not have removed the key from the keypool
+        assert(changeaddress == nextaddr)
+
+        result3 = self.nodes[3].fundrawtransaction(rawtx)
+        res_dec = self.nodes[0].decoderawtransaction(result3["hex"])
+        changeaddress = ""
+        for out in res_dec['vout']:
+            if out['value'] > 1.0:
+                changeaddress += out['scriptPubKey']['addresses'][0]
```
Comment (jtimon) [inline qa/rpc-tests/fundrawtransaction.py]: why += instead of a regular assignment? What if there's more outputs than one with value > 1?

**Comment by jtimon** [comment]
utACK 161af5b modulo the tests which I'm not sure I'm fully understanding. Needs rebase.

**Comment by TheBlueMatt** [comment]
Needs rebase (but is a bugfix, so can go in for 0.14 after the feature freeze on Monday).

**Comment by jonasschnelli** [comment]
Rebased.

**Comment by laanwj** [comment]
Not going to block this on a test nit - that can be fixed later.
I have enough confidence in this to sneak this in before the feature freeze.

**Comment by gituser** [comment]
Just a quick question to confirm: so in the latest master there is no more problem at fundrawtransaction()  with address reuse?

We hit an issue when sending transaction via fundrawtransaction change was sent to the existing user account address and thus credited our user with balance.

sendtoaddress always creates a new change address I suppose.

**Comment by jonasschnelli** [comment]
> We hit an issue when sending transaction via fundrawtransaction change was sent to the existing user account address and thus credited our user with balance.

Yes. This is exactly what this PR does fix.

**Pull request merged**